src/System.Text.Http.Parser/HttpParser.cs

```diff
@@ -42,7 +42,7 @@ public class HttpParser : IHttpParser
             }
             else
             {
-                span = TryGetNewLineSpan(ref buffer, out consumed);
+                span = TryGetNewLineSpan(buffer, out consumed);
                 if (span.Length == 0)
```

Comment:
if buffer is an ordinary parameter, passing it by ref results in a span that cannot be returned.
I made it an "in" in the containing method to widen its escape.
Made it "in" in couple more places. 
ReadableBuffer is a huge struct, I guess that was a reason why it was passed by ref to TryGetNewLineSpan, since otherwise it is no being mutated there.